auth-next/oidc.js

```diff
@@ -0,0 +1,93 @@
+// [SNIPPET_REGISTRY disabled]
+// [SNIPPETS_SEPARATION enabled]
+
+function oidcProvider() {
+  // [START auth_oidc_provider_create]
+  const { OAuthProvider } = require("firebase/auth");
+
+  const provider = new OAuthProvider("oidc.myProvider");
+  // [END auth_oidc_provider_create]
+}
+
+function oidcSignInPopup(provider) {
+  // [START auth_oidc_signin_popup]
+  const { getAuth, signInWithPopup, OAuthProvider } = require("firebase/auth");
+
+  const auth = getAuth();
+  signInWithPopup(auth, provider)
+    .then((result) => {
+      // User is signed in.
+    }).catch((error) => {
+      // Handle Errors here.
+      const errorCode = error.code;
+      const errorMessage = error.message;
+      // The email of the user's account used.
+      const email = error.email;
+      // The AuthCredential type that was used.
+      const credential = OAuthProvider.credentialFromError(error);
+      // ...
```

Comment:
Here and below on L62 and L90, I'd mention here what to do with the error attributes.